### ISSUE TITLE
[Password page] Missing header landmark

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#706]
## Issue summary

The password template is missing its `header` landmark element. Landmarks provide greater semantic meaning and page structure for users of assistive technology. Advanced browsing modes in screen readers will miss out on announcing this information when navigating via landmarks alone.

<details>
  <summary>Screenshots</summary>

  <img width="884" alt="Dawn password page. Heading section including store logo and Enter using password control is highlighted." src="https://user-images.githubusercontent.com/1392632/135288552-2767fb21-10b6-4825-a47a-d11dd2f0c316.png">

</details>

#### HTML

```html
<div id="shopify-section-main-password-header" class="shopify-section">
  <div class="color-background-1 gradient">
    <div class="password-header">
      <h1 class="h2">Store Title</h1>
      <!-- Password modal… -->
    </div>
  </div>
  <hr />
</div>
```

## Steps to reproduce

1. Go to a Dawn theme store with a password enabled with any supported browser
2. Navigate to the `/password` page
3. Use the DevTools accessibility inspector to review the header content markup

### Expected

- Header content to be wrapped with a `header` element.

### Actual

- `header` element is missing.

## Recommendation

Swap the `div#shopify-section-main-password-header` for a `header` element. This will provide the header landmark for assistive technology users.

#### HTML

```html
<header id="shopify-section-main-password-header" class="shopify-section">
  <div class="color-background-1 gradient">
    <div class="password-header">
      <h1 class="h2">Store Title</h1>
      <!-- Password modal… -->
    </div>
  </div>
  <hr />
</header>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.1.0
- **Component:** Password page
- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- **Severity:** Medium
- **Effort:** Low

**Comment by mqnguyen5** [comment]
I would like to work on this. Can you assign the issue to me?

So, to my understanding, I just need to swap the `<div>` with a `<header>` element?
